### ISSUE TITLE
Mappers should return SparsePauliOp instead of PauliSumOp

### ISSUE THREAD

**Issue by kevinsung** [Qiskit/qiskit-nature#879]
### What is the expected enhancement?

See title.

**Comment by mrossinek** [comment]
Thanks for opening this issue. As discussed offline, there is multiple aspects that need to happen here which would be good to describe. Otherwise this issue as such is not that useful.

For context: with the move to the primitives in `qiskit.algorithms` as part of the 0.22 release of Qiskit Terra, we started to phase out the usage of the `qiskit.opflow` module in favor of the `qiskit.quantum_info`. As such, an effort was made to only rely on `PauliSumOp` as the final `opflow` object still being used in the public API.
The reason for not transitioning fully to `quantum_info` is the implementation of the `Z2Symmetries` which has not been ported.
More specifically, there are 2 parts here:
- the general `Z2Symmetries`
- the more specific `TwoQubitReduction`
The latter is very specific to the `ParityMapper` here in Nature so I would suggest to migrate this back to Nature.
The former (afaik) is also only being used by Qiskit Nature, although technically it can also be used elsewhere, if one is able to determine the correct sector of the Hilbert space, in which the desired state lies. A discussion will need to be held, on whether to migrate the `Z2Symmetries` to `quantum_info` or to Nature.

All that said, the importance of these qubit reduction techniques is going to diminish over time, since the removal of a handful of qubits will have less of an impact the larger the total number of qubits (i.e. relatively speaking, the reduction may not be worth the classical overhead). This is however still a long-term aspect to figure out, and for the foreseeable future an implementation of the `Z2Symmetries` is highly desirable.

<hr/>

Once the above has been figured out, Nature will be able to transition to the usage of `SparsePauliOp`s.

**Comment by mrossinek** [comment]
This was just implemented as part of #1050. On the latest main branch (which will eventually be released as 0.6.0), you can now enable this feature already by doing the following:
```python
from qiskit_nature import settings

settings.use_pauli_sum_op = False
```
Afterwards, your mappers will produce `SparsePauliOp` instances instead of `PauliSumOp` ones.
